### ISSUE TITLE
yarn doesn't recognize transitive dependencies between multiple local tarball file installs unlike npm

### ISSUE THREAD

**Issue by mysticflute** [yarnpkg/yarn#7225]
**Do you want to request a *feature* or report a *bug*?**

bug

**What is the current behavior?**
When you have multiple local tarball files to install, and some have dependencies on others then yarn attempts to resolve the dependency from the npm registry instead of the provided local files. When you do this with npm it correctly identifies the dependency.

**If the current behavior is a bug, please provide the steps to reproduce.**

For example, say I have these files:

```
lib/a.tgz
lib/b.tgz
```
`a` has a dependency on `b`. 

When I do the following:
`yarn add lib/a.tgz lib/b.tgz`

it will throw an error such as this:
`Couldn't find package "b" required by "file:lib/a.tgz" on the "npm" registry.`

In other words, even though b.tgz is provided it is trying to resolve it from the npm registry.

However when you try it with npm it works as expected:
`npm install lib/a.tgz lib/b.tgz`

When you look in package-lock.json you see that a and b are both resolved to the local tarballs.

**What is the expected behavior?**
the same behavior as npm, where you can install local tarballs that have a dependency fulfilled by another local tarball.

**Please mention your node.js, yarn and operating system version.**
node: v10.15.3
yarn: 1.15.2
os: macos 10.14.4

**Comment by mysticflute** [comment]
A workaround for this is to use the yarn resolutions field in package.json: https://yarnpkg.com/en/docs/selective-version-resolutions/
